include/zephyr/drivers/fuel_gauge.h

```diff
@@ -215,17 +204,17 @@ struct sbs_gauge_device_chemistry {
  *
  * See fuel_gauge_get_property() for argument description
  */
-typedef int (*fuel_gauge_get_property_t)(const struct device *dev,
-					 struct fuel_gauge_property *prop);
+typedef int (*fuel_gauge_get_property_t)(const struct device *dev, fuel_gauge_prop_t prop_type,
+					 union fuel_gauge_prop_val *prop);
```

Comment:
I'd standardize all of these to `prop` and `val`, same as charger, sensors has like `chan, val` or `attr, val`...